lib/ansible/modules/storage/purestorage/purefa_offload.py

```diff
@@ -0,0 +1,271 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: purefa_offload
+version_added: '2.8'
+short_description: Create, modify and delete NFS or S3 offload targets
+description:
+- Create, modify and delete NFS or S3 offload targets.
+- Only supported on Purity v5.2.0 or higher.
+- You must have a correctly configured offload network for offload to work.
+author:
+- Pure Storage Ansible Tesm (@sdodsley) <pure-ansible-team@purestorage.com>
+options:
+  state:
+    description:
+    - Define state of offload
+    default: present
+    choices: [ absent, present ]
+    type: str
+  name:
+    description:
+    - The name of the offload target
+    required: true
+    type: str
+  protocol:
+    description:
+    - Define which protocol the offload engine uses
+    default: nfs
+    choices: [ nfs, s3 ]
+    type: str
+  address:
+    description:
+    - The IP or FQDN address of the NFS server
+    type: str
+  share:
+    description:
+    - NFS export on the NFS server
+    type: str
+  options:
+    description:
+    - Additonal mount options for the NFS share
+    - Supported mount options include I(port), I(rsize),
+      I(wsize), I(nfsvers), and I(tcp) or I(udp)
+    required: false
+    default: ""
+    type: str
+  access_key:
+    description:
+    - Access Key ID of the S3 target
+    type: str
+  bucket:
+    description:
+    - Name of the bucket for the S3 target
+    type: str
+  secret:
+    description:
+    - Secret Access Key for the S3 target
+    type: str
+  initialize:
+    description:
+    - Define whether to initialize the S3 bucket
+    type: bool
+    default: true
+
+extends_documentation_fragment:
+- purestorage.fa
+'''
+
+EXAMPLES = r'''
+- name: Create NFS offload target
+  purefa_offload:
+    name: nfs-offload
+    protocol: nfs
+    address: 10.21.200.4
+    share: "/offload_target"
+    fa_url: 10.10.10.2
+    api_token: e31060a7-21fc-e277-6240-25983c6c4592
+
+- name: Create S3 offload target
+  purefa_offload:
+    name: s3-offload
+    protocol: s3
+    access_key: "3794fb12c6204e19195f"
+    bucket: offload-bucket
+    secret: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+    fa_url: 10.10.10.2
+    api_token: e31060a7-21fc-e277-6240-25983c6c4592
+
+- name: Delete offload target
+  purefa_offload:
+    name: nfs-offload
+    protocol: nfs
+    state: absent
+    fa_url: 10.10.10.2
+    api_token: e31060a7-21fc-e277-6240-25983c6c4592
+
+
+'''
+
+RETURN = r'''
+'''
+
+import re
+from distutils.version import LooseVersion
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pure import get_system, purefa_argument_spec
+
+MIN_REQUIRED_API_VERSION = '1.16'
```

Comment:
Would this be worth documenting in the module's requirements field?

Reply:
The module documentation declares the minimum version of the array OS required. This is translated through the REST API as the version listed. The user has no visibility of the API version used under the covers.

Reply:
roger that, thanks for the clarification @sdodsley